### PR TITLE
Refactor/docker compose

### PR DESCRIPTION
**Pull request by nogahaskal**
Change docker compose version to 3, and use .env file

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,22 +1,25 @@
-version: '2.1'
+version: "3"
 services:
   spike-service:
+    build: .
     image: spike-service:latest
     ports:
-      - '8080:8080'
+      - "8080:8080"
```
Comment (Shahar-Y) [inline docker-compose.yml:7]: I see you really do hate those single quotes

File: docker-compose.yml [hunk 2]
```diff
@@ -1,22 +1,25 @@
...
+      AUDIENCE: ${KARTOFFEL_AUDIENCE}
+      GRANT_TYPE: ${GRANT_TYPE}
+      SPIKE_CLIENT_ID: ${SPIKE_CLIENT_ID}
+      SPIKE_CLIENT_SECRET: ${SPIKE_CLIENT_SECRET}
+      REDIS_URL: ${REDIS_URL}
+      SPIKE_TOKEN_URL: ${SPIKE_TOKEN_URL}
+      SPIKE_PUBKEY_PATH: ${SPIKE_PUBKEY_PATH}
+      SPIKE_PUBLIC_KEY_FULL_PATH: ${SPIKE_PUBLIC_KEY_FULL_PATH}
+      NODE_TLS_REJECT_UNAUTHORIZED: ${NODE_TLS_REJECT_UNAUTHORIZED}
+      SS_PROT: ${PORT}
```
Comment (Shahar-Y) [inline docker-compose.yml:18]: 🦜 
Also, 'port' seems too generic for an .env file which will contain envars for a lot of services.

File: docker-compose.yml [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
+      SS_HOST: ${HOST}
```
Comment (Shahar-Y) [inline docker-compose.yml:19]: image
Is SS the prefix for search service or spike?
Also, 'host' seems too generic for an .env file which will contain envars for a lot of services.

File: docker-compose.yml (hunk 1 above)
Comment (nogahaskal) [reply docker-compose.yml:7]: i do.

File: docker-compose.yml (hunk 2 above)
Comment (nogahaskal) [reply docker-compose.yml:18]: it is so generic because it used in all the services haha (8080)

File: docker-compose.yml (hunk 3 above)
Comment (nogahaskal) [reply docker-compose.yml:19]: same as the other comment. 
the prefix is for spike, i added that because it's in the config but never used.

Comment (nogahaskal) [reply docker-compose.yml:19]: also, hyle hitler

File: docker-compose.yml (hunk 2 above)
Comment (Shahar-Y) [reply docker-compose.yml:18]: So I'd call it `SERVICE_PORT` or something of this sort

File: docker-compose.yml (hunk 3 above)
Comment (Shahar-Y) [reply docker-compose.yml:19]: Fisrst, heil*. Very important.
Second, the meaning of the wotd HOST alone is not clear, so add another word to the name describing it better.
Third, the prefix in the other envars was SPIKE, Why not keep it that way?

Comment (Shahar-Y) [reply docker-compose.yml:19]: Regarding `Third` - I got it now. I blame @AllKinds .

**Pull request merged**